apps/src/code-studio/initApp/clientApi.js

```diff
@@ -201,7 +202,11 @@ var base = {
         callback(null, data);
       })
       .fail(function (request, status, error) {
-        var err = errorString(request, status, error);
+        const args = [request, status, error];
+        if (opts.includeDetails && request.responseJSON?.details) {
+          args.push(request.responseJSON?.details);
+        }
+        var err = errorString(...args);
```

Comment:
This is such a weird way to do this, but I can't think of a better way, given how we use this `errorString` function everywhere. Any ideas?

Reply:
One idea (this might be kind of half baked)...what if you passed the bad word from the server as an object (eg, {content: 'badword'}), then stored it as the `cause` (optional second argument to JS Error constructor)? Seems like a valid use of that argument:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

Then maybe you wouldn't have to the regex in the React component (and my other comment is probably moot).

Reply:
Updated!